source/extensions/filters/network/common/redis/fault.cc

```diff
@@ -0,0 +1,122 @@
+#include "extensions/filters/network/common/redis/fault.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace Common {
+namespace Redis {
+
+  RedisFaultManager::RedisFaultManager(Runtime::RandomGenerator& random, Runtime::Loader& runtime, const ::google::protobuf::RepeatedPtrField< ::envoy::extensions::filters::network::redis_proxy::v3::RedisProxy_RedisFault> faults) : random_(random), runtime_(runtime) {
+    // Group faults by command
+    for (auto fault : faults) {
+      if (fault.commands_size() > 0) {
+        for (auto& command: fault.commands()) {
+          std::cout << "Inserting fault for command: " << absl::AsciiStrToLower(command) << " with type: " << fault.GetTypeName() << std::endl;
+          fault_map_.insert(FaultMapType::value_type(absl::AsciiStrToLower(command), fault));
+        }
+      } else {
+        // Generic "ALL" entry in map
+        std::cout << "Inserting fault for all: " << ALL_KEY << std::endl;
+        fault_map_.insert(FaultMapType::value_type(ALL_KEY, fault));
+      }
+    }
+  }
+
+  absl::optional<std::pair<FaultType, std::chrono::milliseconds>> RedisFaultManager::get_fault_for_command(std::string command) {
+    // Fault checking algorithm:
+    //
+    // For example, if we have an ERROR fault at 5% for all commands, and a DELAY fault at 10% for GET, if
```

Comment:
This algorithm doesn't seem right.  If we already have the list of faults and their %, then why can't we just get a random number between 0 and 1, [0-0.05) would be ERROR, [0.05-0.1) would be DELAY, [0.1-1] would be no fault.

Reply:
We could do that, but we still need to iterate over all the faults and get their percentage as the runtime value may have changed.